utilities/qneg.F90

```diff
@@ -0,0 +1,388 @@
+module qneg
+
+  use shr_kind_mod,        only: CS => SHR_KIND_CS
+  use ccpp_kinds,          only: kind_phys
+  use perf_mod,            only: t_startf, t_stopf
+  use cam_logfile,         only: iulog
+  use shr_sys_mod,         only: shr_sys_flush
+!  use cam_history_support, only: max_fieldname_len
+!  use ppgrid,              only: pcols
+
+  implicit none
+  private
+  save
+
+  ! Public interface.
+
+  public :: qneg_init
+  public :: qneg_run !qneg3 in CAM6
+  public :: qneg_timestep_final
+  public :: qneg_final
+
+  ! Private module variables
+  logical          :: collect_stats = .false.
+  logical          :: timestep_reset = .false.
+  logical          :: qneg_initialized = .false.
+  integer          :: num_constituents = 0
+
+  real(kind_phys), parameter :: tol = 1.e-12_kind_phys
+  real(kind_phys), parameter :: worst_reset = 1.e35_kind_phys
+
+  ! Diagnostic field names
+!  integer              :: num_diag_fields = 0
+!  character(len=max_fieldname_len) :: diag_names(num_diag_fields)
+!  logical             :: cnst_out_calc = .false.
+!  logical             :: cnst_outfld(num_diag_fields) = .false.
+
+  ! Summary buffers
+  integer, parameter    :: num_bins = 24
```

Comment:
added a comment. Not sure about the first question - @nusbaume ?

Reply:
Assuming the number of schemes that call qneg is equal to the number of qneg calls in a particular SDF then determining the `num_bins` value ahead of time and passing it in should totally be doable.   That being said, although we could get CAMDEN to do it, I think it would be a lot cleaner and more portable if the CCPP framework itself managed it (as @cacraigucar mentioned).  Maybe we should make another issue in this repo to remind us to ask Steve when he gets back?

Reply:
Issue - https://github.com/NCAR/atmospheric_physics/issues/41